framework/web/Response.php

```diff
@@ -362,8 +362,8 @@ public function clear()
      */
     protected function sendHeaders()
     {
-        if (headers_sent()) {
-            return;
+        if (headers_sent($file, $line)) {
+            throw new HeadersAlreadySentException($file, $line);
```

Comment:
This is a serious BC break - disallows setting flexibly custom headers anywhere in the code, e.g. when downloading files.

Reply:
You were not able to set them anyway. It was just silently ignored.

Reply:
We used to set various custom headers directly e.g. 
`header('Content-Type: application/octet-stream', false);`
Now everybody who used to set inline headers will get Exception. With already 14th branch release there might be heaps of code to be refactored :-( It would be better to leave such a BC change for 2.1.X. I am not sure what you mean `not able to set them anyway` - it was possible via `$response->getHeaders()->add()`, right? But we did not use it coz it's sometime easier to write native PHP headers with second `replace` option.

Reply:
If there's now an exception then these headers weren't working well before.